source/extensions/tracers/xray/tracer.h

```diff
@@ -107,62 +110,77 @@ class Span : public Tracing::Span {
   Envoy::SystemTime startTime() const { return start_time_; }
 
   /**
-   * Get the end time of this Span.
+   * Get this Span's ID.
    */
-  Envoy::SystemTime endTime() const { return end_time_; }
+  const std::string& Id() const { return id_; }
+
+  const std::string& parentId() const { return parent_segment_id_; }
 
   /**
-   * Get this Span's ID.
+   * Get this Span's name.
    */
-  uint64_t Id() const { return id_; }
+  const std::string& name() const { return name_; }
+
+  /**
+   * Determine whether this span is sampled.
+   */
+  bool sampled() const { return sampled_; }
 
   /**
    * Not used by X-Ray because the Spans are "logged" (serialized) to the X-Ray daemon.
    */
   void log(Envoy::SystemTime, const std::string&) override {}
 
   /**
-   * Not supported by X-Ray.
+   * Create a child span.
+   * In X-Ray terms this creates a sub-segment and sets its parent ID to the current span's ID.
+   * @param operation_name The span of the child span.
+   * @param start_time The time at which this child span started.
    */
-  Tracing::SpanPtr spawnChild(const Tracing::Config&, const std::string&,
-                              Envoy::SystemTime) override {
-    return nullptr;
-  }
+  Tracing::SpanPtr spawnChild(const Tracing::Config&, const std::string& operation_name,
+                              Envoy::SystemTime start_time) override;
 
 private:
-  void setParentSpan(Tracing::Span* parent) { parent_span_ = parent; }
-
-  TimeSource& time_source_;
   Envoy::SystemTime start_time_;
-  Envoy::SystemTime end_time_;
   std::string operation_name_;
+  std::string id_;
   std::string trace_id_;
   std::string parent_segment_id_;
   std::string name_;
-  using Annotation = std::pair<std::string, std::string>;
-  absl::flat_hash_map<std::string, std::vector<Annotation>> annotations_;
-  std::vector<std::pair<std::string, std::string>> metadata_;
-  Tracing::Span* parent_span_;
-  uint64_t id_;
+  absl::flat_hash_map<std::string, std::string> http_request_annotations_;
+  absl::flat_hash_map<std::string, std::string> http_response_annotations_;
+  absl::flat_hash_map<std::string, std::string> custom_annotations_;
+  Envoy::TimeSource& time_source_;
+  DaemonBroker& broker_;
+  bool sampled_;
 };
 
+using SpanPtr = std::unique_ptr<Span>;
+
 class Tracer {
 public:
-  Tracer(absl::string_view segment_name, TimeSource& time_source)
-      : segment_name_(segment_name), time_source_(time_source) {
-    UNREFERENCED_PARAMETER(time_source_);
-  }
+  Tracer(absl::string_view segment_name, DaemonBrokerPtr daemon_broker, TimeSource& time_source)
+      : segment_name_(segment_name), daemon_broker_(std::move(daemon_broker)),
+        time_source_(time_source) {}
 
   /**
-   * Starts a tracing span for X-Ray
+   * Start a tracing span for X-Ray
```

Comment:
The prior spelling is slightly more common I think, but up to you.

Reply:
I want it to be consistent across the API.
I'm ok to change them all. WDYT?

Reply:
My guess is that a significant percentage of Envoy would consistently use "Starts a tracing span..." and the rest would be random. So if you want to be consistent with Envoy style I'd suggest that.

But as I said it's up to you.